src/vsock/README.md

```diff
@@ -0,0 +1,87 @@
+# vhost-user-vsock
+
+## Design
+
+TODO: This section should have a high-level design of the crate.
+
+Some questions that might help in writing this section:
+- What is the purpose of this crate?
+- What are the main components of the crate? How do they interact which each
+  other?
+
+## Usage
+
+Run the vhost-user-vsock device:
+```
+vhost_user_vsock --vsock-backend guest-cid=4,uds-path=/tmp/vm4.vsock,socket=/tmp/vhost4.socket
+```
+
+Run qemu:
+
+```
+qemu-system-x86_64 -drive file=/path/to/disk.qcow2 -enable-kvm -m 512M \
+  -smp 2 -vga virtio -chardev socket,id=char0,reconnect=0,path=/tmp/vhost4.socket \
+  -device vhost-user-vsock-pci,chardev=char0 \
+  -object memory-backend-file,share=on,id=mem,size="512M",mem-path="/dev/hugepages" \
+  -numa node,memdev=mem -mem-prealloc
+```
+
+### Guest listening
+
+#### iperf
+
+```sh
+# https://github.com/stefano-garzarella/iperf-vsock
+guest$ iperf3 --vsock -s
+host$  iperf3 --vsock -c /tmp/vm4.vsock
+```
+
+#### netcat
+
+```sh
+# https://github.com/stefanha/nc-vsock
```

Comment:
Recent `nc` should have vsock support too: https://man7.org/linux/man-pages/man1/ncat.1.html#AF_VSOCK_SOCKETS